packages/artifact/src/artifact.ts

```diff
@@ -0,0 +1,111 @@
+import * as core from '@actions/core'
+import {CreateArtifactResponse} from './contracts'
+import {SearchResult, findFilesToUpload} from './search'
+import {
+  createArtifactInFileContainer,
+  uploadArtifactToFileContainer,
+  patchArtifactSize
+} from './upload-artifact-http-client'
+import {UploadInfo} from './upload-info'
+import {checkArtifactName} from './utils'
+
+/**
+ * Uploads an artifact
+ *
+ * @param name the name of the artifact, required
+ * @param path the directory, file, or glob pattern to denote what will be uploaded, required
+ * @returns single UploadInfo object
+ */
+export async function uploadArtifact(
+  name: string,
+  path: string
+): Promise<UploadInfo> {
+  // Check that the required inputs are valid
+  if (!name) {
+    throw new Error('Artifact name must be provided')
+  }
+
+  checkArtifactName(name)
+
+  if (!path) {
+    throw new Error('Upload path must be provided')
+  }
+
+  // Search for the items that will be uploaded
+  let filesToUpload: SearchResult[] = await findFilesToUpload(name, path)
+
+  if (filesToUpload === undefined) {
+    core.setFailed(
+      `Unable to succesfully search fo files to upload with the provided path: ${path}`
+    )
+  } else if (filesToUpload.length == 0) {
+    core.warning(
+      `No files were found for the provided path: ${path}. No artifacts will be uploaded.`
+    )
+  } else {
+    /**
+     * Step 1 of 3
+     * Create an entry for the artifact in the file container
+     */
+    const response: CreateArtifactResponse = await createArtifactInFileContainer(
+      name
+    )
```

Comment:
Single line with `;`?

Reply:
This was automatically changed by `prettier`. If I change it to a single line it complains: 
![image](https://user-images.githubusercontent.com/16109154/72358142-16a5c100-36ba-11ea-86fd-309dcea3ee6e.png)

Reply:
Was able to get it shortened in a single line without `prettier` complaining